File: commons/src/main/java/com/nerdscorner/mvplib/commons/behaviour/Attachable.java
package com.nerdscorner.mvplib.commons.behaviour;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

/**
 * Public interface for attaching specific contexts
 */
public interface Attachable {

    /**
     * Set this behaviour's execution context.
     *
     * @param activity the view where this behaviour is attached.
     */
    void attach(@NonNull final AppCompatActivity activity);

    /**
     * Set this behaviour's execution context.
     *
     * @param fragment the view where this behaviour is attached.
     */
    void attach(@NonNull final Fragment fragment);

}

File: commons/src/main/java/com/nerdscorner/mvplib/commons/behaviour/Attachable.kt
package com.nerdscorner.mvplib.commons.behaviour

import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity

/**
 * Public interface for attaching specific contexts
 */
interface Attachable {

    /**
     * Set this behaviour's execution context.
     *
     * @param activity the view where this behaviour is attached.
     */
    fun attach(activity: AppCompatActivity)

    /**
     * Set this behaviour's execution context.
     *
     * @param fragment the view where this behaviour is attached.
     */
    fun attach(fragment: Fragment)

}
